client/src/components/Profile/CommonCard.tsx

```diff
@@ -1,114 +1,101 @@
-import * as React from 'react';
+import { ReactNode, useState } from 'react';
 import { Card, Typography, Empty } from 'antd';
-import { ConfigurableProfilePermissions } from 'common/models/profile';
-import PermissionsSettingsDrawer from './PermissionsSettingsDrawer';
-import ProfileSettingsDrawer from './ProfileSettingsDrawer';
-import { ChangedPermissionsSettings } from 'pages/profile';
-import { CheckboxChangeEvent } from 'antd/lib/checkbox';
+import { EditOutlined } from '@ant-design/icons';
+import ProfileSettingsModal from './ProfileSettingsModal';
 
 const { Title } = Typography;
 
-import { EditOutlined, SettingOutlined } from '@ant-design/icons';
-
 type Props = {
-  permissionsSettings?: Partial<ConfigurableProfilePermissions>;
   profileSettingsContent?: JSX.Element;
   isEditingModeEnabled?: boolean;
-  noDataDescrption?: string;
+  noDataDescription?: string;
   settingsTitle?: string;
   title: string;
-  icon: any;
-  content: any;
-  actions?: any;
-  onPermissionsSettingsChange?: (event: CheckboxChangeEvent, settings: ChangedPermissionsSettings) => void;
+  icon: JSX.Element;
+  content: JSX.Element | null;
+  actions?: ReactNode[];
+  isSaveDisabled?: boolean;
   detachSettingsOnVisibilityChange?: boolean;
+  saveProfile?: () => void;
+  onCancel?: () => void;
 };
 
-type State = {
-  isVisibilitySettingsVisible: boolean;
-  isProfileSettingsVisible: boolean;
-};
-class CommonCard extends React.Component<Props, State> {
-  state = {
-    isVisibilitySettingsVisible: false,
-    isProfileSettingsVisible: false,
-  };
+const CommonCard = ({
+  title,
+  icon,
+  content,
+  profileSettingsContent,
+  isEditingModeEnabled,
+  noDataDescription,
+  settingsTitle,
+  actions,
+  detachSettingsOnVisibilityChange,
+  isSaveDisabled,
+  saveProfile,
+  onCancel: cancelChanges,
+}: Props) => {
+  const [isProfileSettingsVisible, setIsProfileSettingsVisible] = useState(false);
+  const renderSettingsModal = detachSettingsOnVisibilityChange ? isProfileSettingsVisible : true;
```

Comment:
It would be better to rename this variable to avoid mess with functions

Reply:
Agreed, thanks.